src/FolderConfig.cpp

```diff
@@ -0,0 +1,94 @@
+#include "FolderConfig.h"
+#include <QFile>
+#include <QDebug>
+#include<QDir>
+
+namespace QtDBMigrationNS {
+FolderConfig::FolderConfig(const QString &path): AbstractConfig(path)
+{
+}
+
+bool FolderConfig::load()
+{
+    QDir dir(m_path);
+    QStringList nameFilter("*.sql");
+    QStringList sqlFiles = dir.entryList(nameFilter);
+    int totalFiles = sqlFiles.count();
+
+    if (!dir.exists() || totalFiles < 1) {
+        return false;
+    }
+
+    for(auto filename : sqlFiles) {
+        bool migrationsUp = true;
+        QString filePath = dir.filePath(filename);
+        QFile file(filePath);
+        if (!file.exists()) {
+            return false;
+        }
+
+        QFileInfo fileInfo(file.fileName());
+        QRegExp rx("^\\d{4}\\.sql$");
+        if (!rx.exactMatch(fileInfo.fileName())) {
+            continue;
+        }
+
+        if(!file.open(QFile::ReadOnly | QIODevice::Text)) {
+            return false;
+        }
+
+        QTextStream in(&file);
+        QString line = in.readLine();
+        QStringList applySql;
+        QStringList revertSql;
+        while (!line.isNull())
+        {
+            if (line.size()==0)
+            {
+                line = in.readLine();
+                continue;
+            }
+
+            if (line.contains("-- +"))
```

Comment:
Maybe use `QRegExp` instead?

Something in line `"^-- \+migrate (?P<operation>Up|Down)"`. And then you can use the captured text `operation`. Will make some ifs go away.

And also, code more readable and more robust.

Reply:
Changed to using ```QRegularExpression```.